packages/utils/feed/markdown.ts

```diff
@@ -0,0 +1,227 @@
+import { Element as DomHandlerElement } from "domhandler";
+import markdownit from "markdown-it";
+import { full as emoji } from "markdown-it-emoji/dist/index.cjs";
+import footnote_plugin from "markdown-it-footnote";
+import { MixedStyleRecord } from "react-native-render-html";
+
+import {
+  neutral17,
+  neutral33,
+  neutral67,
+  neutralA3,
+  neutralFF,
+  primaryColor,
+} from "@/utils/style/colors";
+
+export type ContentMode = "EDITION" | "BOTH" | "PREVIEW";
+
+// The markdownit instance. Used to get the same parameters at Article creation and consultation.
+export const articleMd = markdownit({
+  linkify: true,
+  breaks: true,
+})
+  .use(emoji)
+  .use(footnote_plugin);
+
+// DOM modifications on document, texts, or elements from react-native-render-html.
+// Because react-native-render-html doesn't allow common CSS selectors, we need to style tags using domVisitors callbacks
+export const renderHtmlDomVisitors = {
+  onElement: (element: DomHandlerElement) => {
+    // Removing marginBottom from the child p of blockquote
+    if (
+      element.name === "blockquote" &&
+      element.children &&
+      element.children.length > 0
+    ) {
+      const tagChild = element.children.find((child) => child.type === "tag");
+      // tagChild is a domhandler Node. It doesn't have attribs, but it has attribs in fact (wtf ?)
+      if (tagChild && "attribs" in tagChild) {
+        tagChild.attribs = {
+          style: "margin-bottom: 0",
+        };
+      }
+    }
+  },
+};
+
+// HTML tags styles used by RenderHtml from react-native-render-html
+export const renderHtmlTagStyles: MixedStyleRecord = {
```

Comment:
<img width="839" alt="Screenshot 2024-12-29 at 13 34 54" src="https://github.com/user-attachments/assets/fb642948-fee1-43ca-88b4-1952089637db" />

is it expected that the h1 has a space on top?

Reply:
Yes, the preview takes more space, we can't afford that
![image](https://github.com/user-attachments/assets/4d323873-7a4e-40c0-8ca6-3fcaf6962595)

headings vertical spaces
![image](https://github.com/user-attachments/assets/3049fd71-5a05-4229-9d0a-a1e4f44aa245)

top space to compensate the input borders padding
![image](https://github.com/user-attachments/assets/cc0d1a5e-70b2-49b1-a18a-4f1be386d4ee)

The goal is to having the same vertical and horizontal spaces in the edition et the preview modes
![image](https://github.com/user-attachments/assets/26fca9a9-92cc-4533-9f1e-0e218c67212e)